### DISCUSSION TITLE
Patreon OAuth API

### DISCUSSION THREAD

**Discussion by Decollates** [gameserverapp/Platform#120] [Suggestions]
The ability to 1: Connect the community patreon to GSA's database, and 2: allow for users to connect their own accounts (similar to how they can authenticate their steam, twitch, and discord accounts via the community website). 

Tasks/triggers can then be made to activate in order to execute certain commands, deliver shop packs, and apply roles for users. The biggest benefit for this would be the automation of dispensing rewards for community members who support their community via patreon, with the community being otherwise unable to utilize GSA's subscriber function.

**Comment by mvaessen** [comment]
Created development item #122. If you have any further suggestions, please add to the development item.

**Comment by UpFromTheDepths** [comment]
YES PLEASE! Majority of my funding comes through Patreon and it's a bit of a hassle giving people rewards every month or so, often times I do forget and peeps have to remind me lol. I would certainly welcome Patreon integration!

**Comment by mvaessen** [comment]
I spent some time thinking about how this would work best for both the admin team and the players.

How do you feel about having a special Patreon Supporter Tier?
This would work pretty much like how PayPal works, but with Patreon instead. Development wise this should be easier to implement vs a task-oriented setup.

**Reply by UpFromTheDepths** [reply]
That works too!

**Reply by Decollates** [reply]
Just seeing this--

How would a patreon-specific Support Tier work, exactly?

**Reply by mvaessen** [reply]
When people connect their Patreon, it will check for renewals. When the patreon renews, people get the associated supporter tier. When their patreon expires, the "expire" actions run for the supporter tier.

This is with the assumption that patreon does support these kind of things in their webhook systems.

**Reply by Decollates** [reply]
I like this. I was gonna make a suggestion for certain tasks to run at the end of the month if Patreon support ever got implemented (such as running RCON commands to remove Permissions-based plugin roles). My patreon is set to charge at sign-up and on the first of each month, but typically players are allowed to 'keep' their privileges until the end of the month if they cancel early. For example, someone subs on the 3rd, but cancels by the 15th, they can still enjoy the patreon benefits until the end of the month. Would this system be able to handle that?

**Reply by mvaessen** [reply]
That sounds very similar to how recurring supporter tiers work at the moment with PayPal. Most likely it will work like that.

**Reply by Decollates** [reply]
Wonderful!
